### ISSUE TITLE
how to get the correct file type in the input event,  uploadFile's fileType by puppeteer's uploadFile api

### ISSUE THREAD

**Issue by TimCN** [puppeteer/puppeteer#5443]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 2.1.1
* Chrome Version: 80.0.3987.116
* Platform / OS version: macOS Mojave Version 10.14.6
* URLs (if applicable):
* Node.js version: 12.14.1

**What steps will reproduce the problem?**

_Please include code that reproduces the issue._

1. site code
```jsx
const ImageUploader = () => {


const onFileInput = (e) => {
   const fileType = e.target.files[0].type
   if(fileType.indexOf("image/") !== 0) {
      window.alert("image only");
      return
   } else {
     // todo 
   }
}

return <input type="file" onInput={onFileInput} />

}
```

2. test code
```js
const puppeteer = require("puppeteer-core")
const browser = await puppeteer.launch({
executablePath: "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
})
// ...
const page = await browser.newPage();
await page.goto("my_private_project_page_path");
const uploader = await page.$("input[type=file]");
await uploader.uploadFile("a_jpeg_image_path")
```
3. normally, I upload a image can never get the "image only" alert, until by puppeteer's uploadFile

**What is the expected result?**

I can get the correct file type in the input event, when i upload an image by puppeteer's uploadFile api

**What happens instead?**

No idea
Please help me guys

**Comment by TimCN** [comment]
i try to read puppeteer-core's source code under node_modules, found that the uploadFile api(in puppeteer-core/lib/JSHandle.js) forget pass the file type when new a File instance.and i try to fixed it like below, it works ok on my local
```js
await this.evaluateHandle(async(element, files) => {
      const dt = new DataTransfer();
      for (const item of files) {
        const response = await fetch(`data:${item.mimeType};base64,${item.content}`);
        // const file = new File([await response.blob()], item.name);
        const file = new File([await response.blob()], item.name,{type:item.mimeType});
        dt.items.add(file);
      }
      element.files = dt.files;
      element.dispatchEvent(new Event('input', { bubbles: true }));
    }, files);
```

**Comment by svycka** [comment]
I have the same problem would be nice if this could be fixed

**Comment by matheusb-comp** [comment]
@TimCN Thank you! I also had a problem with this and adding the options object with the `type` property when creating the file works.

Another detail is that in my case, the page has no listener for the `input` event, and I must send the `change` event:
```
element.dispatchEvent(new Event('change', { bubbles: true });
```

**Comment by paullewis** [comment]
This should be fixed as of 532ae573d29063ffd081ed9ee340d71b25320dec as we changed the underlying implementation to use `DOM.setFileInputFiles`, which behaves a lot more as expected. Going to close this issue, but please let us know if it's not fixed.
